Mathlib/CategoryTheory/Triangulated/Opposite.lean

```diff
@@ -300,8 +316,103 @@ lemma isomorphic_distinguished (T₁ : Triangle Cᵒᵖ)
+  rw [mem_distinguishedTriangles_iff']
+  exact ⟨_, inv_rot_of_dist_triangle _ (Pretriangulated.contractible_distinguished X.unop),
+    ⟨contractibleTriangleIso X⟩⟩
+
+/-- Isomorphism expressing a compatibility of the equivalence `triangleOpEquivalence C`
+with the rotation of triangles. -/
+noncomputable def rotateTriangleOpEquivalenceInverseObjRotateUnopIso (T : Triangle Cᵒᵖ) :
+    ((triangleOpEquivalence C).inverse.obj T.rotate).unop.rotate ≅
+      ((triangleOpEquivalence C).inverse.obj T).unop :=
+  Triangle.isoMk _ _ (Iso.refl _) (Iso.refl _)
+      (-((opShiftFunctorEquivalence C 1).unitIso.app T.obj₁).unop) (by simp)
+        (Quiver.Hom.op_inj (by aesop_cat)) (by aesop_cat)
+
+lemma rotate_distinguished_triangle (T : Triangle Cᵒᵖ) :
+    T ∈ distinguishedTriangles C ↔ T.rotate ∈ distinguishedTriangles C := by
+  simp only [mem_distinguishedTriangles_iff, Pretriangulated.rotate_distinguished_triangle
+    ((triangleOpEquivalence C).inverse.obj (T.rotate)).unop]
+  exact distinguished_iff_of_iso (rotateTriangleOpEquivalenceInverseObjRotateUnopIso T).symm
+
+lemma distinguished_cocone_triangle {X Y : Cᵒᵖ} (f : X ⟶ Y) :
+    ∃ (Z : Cᵒᵖ) (g : Y ⟶ Z) (h : Z ⟶ X⟦(1 : ℤ)⟧),
+      Triangle.mk f g h ∈ distinguishedTriangles C := by
+  obtain ⟨Z, g, h, H⟩ := Pretriangulated.distinguished_cocone_triangle₁ f.unop
+  refine' ⟨_, g.op, (opShiftFunctorEquivalence C 1).counitIso.inv.app (Opposite.op Z) ≫
+    (shiftFunctor Cᵒᵖ (1 : ℤ)).map h.op, _⟩
+  simp only [mem_distinguishedTriangles_iff]
+  refine' Pretriangulated.isomorphic_distinguished _ H _ _
+  exact Triangle.isoMk _ _ (Iso.refl _) (Iso.refl _) (Iso.refl _) (by aesop_cat) (by aesop_cat)
+    (Quiver.Hom.op_inj (by simp [shift_unop_opShiftFunctorEquivalence_counitIso_inv_app]))
+
+lemma complete_distinguished_triangle_morphism (T₁ T₂ : Triangle Cᵒᵖ)
+    (hT₁ : T₁ ∈ distinguishedTriangles C) (hT₂ : T₂ ∈ distinguishedTriangles C)
+    (a : T₁.obj₁ ⟶ T₂.obj₁) (b : T₁.obj₂ ⟶ T₂.obj₂) (comm : T₁.mor₁ ≫ b = a ≫ T₂.mor₁) :
+    ∃ (c : T₁.obj₃ ⟶ T₂.obj₃), T₁.mor₂ ≫ c = b ≫ T₂.mor₂ ∧
+      T₁.mor₃ ≫ a⟦1⟧' = c ≫ T₂.mor₃ := by
+  rw [mem_distinguishedTriangles_iff] at hT₁ hT₂
+  obtain ⟨c, hc₁, hc₂⟩ :=
+    Pretriangulated.complete_distinguished_triangle_morphism₁ _ _ hT₂ hT₁
+      b.unop a.unop (Quiver.Hom.op_inj comm.symm)
+  dsimp at c hc₁ hc₂
+  replace hc₂ := ((opShiftFunctorEquivalence C 1).unitIso.hom.app T₂.obj₁).unop ≫= hc₂
+  dsimp at hc₂
+  simp only [assoc, Iso.unop_hom_inv_id_app_assoc] at hc₂
+  refine' ⟨c.op, Quiver.Hom.unop_inj hc₁.symm, Quiver.Hom.unop_inj _⟩
+  apply (shiftFunctor C (1 : ℤ)).map_injective
+  rw [unop_comp, unop_comp, Functor.map_comp, Functor.map_comp,
+    Quiver.Hom.unop_op, hc₂, ← unop_comp_assoc, ← unop_comp_assoc,
+    ← opShiftFunctorEquivalence_unitIso_inv_naturality]
+  simp
+
+/-- The pretriangulated structure on the opposite category of
+a pretriangulated category. -/
+scoped instance : Pretriangulated Cᵒᵖ where
```

Comment:
Why does this instance need to be scoped? Can you please document that here, or add a link to some library note that explains it?

Reply:
Thanks for the review. I have added a comment.